Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,356 @@
+    }
+
+    cruiseSetBitmap &= BUTTON_BITMASK; //selecting the last n bits
+
+    if(cruiseSetBitmap == BUTTON_BITMASK){
+        cruiseSet = true;
+    }else if(cruiseEnableBitmap != BUTTON_BITMASK){
+        cruiseSet = false;
+    }
+
+    // TOGGLE
+    if(ChargeEnable && regenButton != regenPrevious && regenPrevious){regenToggle = !regenToggle;}
+    regenPrevious = regenButton;
+    
+    if(cruiseEnableButton != cruiseEnablePrevious && cruiseEnablePrevious){cruiseEnable = !cruiseEnable;}
+    cruiseEnablePrevious = cruiseEnableButton;
+
+    // LEFT SHIFT
+    cruiseEnableBitmap <<= 1; 
+    regenBitmap <<= 1;
+    cruiseSetBitmap <<= 1;
+}
+
+/**
+ * @brief Converts integer percentage to float percentage
+ * @param percent integer percentage from 0-100
+*/
+extern const float pedalToPercent[];
+static float percentToFloat(uint8_t percent){
+    if(percent > 100){
+        return 1.0f;
+    }
+    return pedalToPercent[percent];
+}
+
+/**
+ * @brief Follows the FSM to update the velocity of the car
+*/
+void Task_SendTritium(void *p_arg){
+    OS_ERR err;
+    Minion_Error_t minion_err;
+    
+    while(1){
+        FSM[state].stateHandler();    // do what the current state does
+        readInputs();   // read inputs from the system
+        FSM[state].stateDecider();    // decide what the next state is
+
+        // Delay of few milliseconds (100)
+        OSTimeDlyHMSM(0, 0, 0, 100, OS_OPT_TIME_HMSM_STRICT, &err);
+        if (err != OS_ERR_NONE){
+            assertOSError(OS_UPDATE_VEL_LOC, err);
+        }
+        // Sets brakelight
+        Minion_Write_Output(BRAKELIGHT, brakelightState, &minion_err);
+    }
+}
+
+// State Handlers & Deciders
+
+/**
+ * Normal Drive State. Accelerator is mapped directly to current setpoint.
+*/
+void NormalDriveHandler(){
+    if(reverseSwitch && !forwardSwitch){
+        velocitySetpoint = -MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }else if(!reverseSwitch && forwardSwitch){
+        velocitySetpoint = MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }
+    else if(!reverseSwitch && !forwardSwitch){
+        currentSetpoint = 0;
+    }
+    brakelightState = false;
+}
+
+void NormalDriveDecider(){
+    if(cruiseSet && cruiseEnable && brakePedalPercent == 0){
+        state = RECORD_VELOCITY;
+    }else if(!cruiseEnable && regenToggle && ChargeEnable){
+        state = ONEPEDAL_DRIVE;
+    }else if(brakePedalPercent > 0){
+        state = BRAKE_STATE;
+    }
+}
+
+/**
+ * Record Velocity State. While pressing the cruise set button, the car will record the observed velocity
+ * into velocitySetpoint.
+*/
+void RecordVelocityHandler(){
+    velocitySetpoint = MotorController_ReadVelocity();
+    brakelightState = true;
+}
+
+void RecordVelocityDecider(){
+    if(cruiseEnable && !cruiseSet){
+        state = POWERED_CRUISE;
+    }else if(!cruiseEnable){
+        state = NORMAL_DRIVE;
+    }else if(brakePedalPercent > 0){
```

Comment:
Shouldn't this be the first if? No matter what, we want to go to the brake state if the brake pedal is pressed.